pkg/api/mock/mock_api_implementation.go

```diff
@@ -0,0 +1,239 @@
+package mock
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/onsi/gomega/ghttp"
+	uuid "github.com/satori/go.uuid"
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+	"go.anx.io/go-anxcloud/pkg/client"
+)
+
+const (
+	// CloudDNS
+	resourcePrefixCloudDNSZone = "/api/clouddns/v1/zone.json"
+
+	// LBaaS
+	resourcePrefixLBaaSLoadBalancer = "/api/LBaaS/v1/loadbalancer.json"
+	resourcePrefixLBaaSFrontend     = "/api/LBaaS/v1/frontend.json"
+	resourcePrefixLBaaSBackend      = "/api/LBaaS/v1/backend.json"
+	resourcePrefixLBaaSServer       = "/api/LBaaS/v1/server.json"
+	resourcePrefixLBaaSBind         = "/api/LBaaS/v1/bind.json"
+	resourcePrefixLBaaSACL          = "/api/LBaaS/v1/ACL.json"
+
+	// VLAN
+	resourcePrefixVlan = "/api/vlan/v1/vlan.json"
+)
+
+type MockAPI interface {
+	api.API
+
+	OperationPathRequestCount(op types.Operation, path string) int
+}
+
+type mockAPI struct {
+	defaultAPI                   api.API
+	mockServer                   *ghttp.Server
+	resourceStorage              map[string]map[string][]byte
+	requestCountForMethodAndPath map[string]int
+	allowNotImplementedPrefixes  bool
+}
+
+func NewMockAPI(allowNotImplemented bool) (MockAPI, error) {
+	mockServer := ghttp.NewServer()
+
+	a, err := api.NewAPI(api.WithClientOptions(
+		client.BaseURL(mockServer.URL()),
+		client.IgnoreMissingToken(),
+	))
+	if err != nil {
+		return nil, err
+	}
+	api := &mockAPI{
+		defaultAPI:                   a,
+		mockServer:                   mockServer,
+		resourceStorage:              make(map[string]map[string][]byte),
+		requestCountForMethodAndPath: make(map[string]int),
+		allowNotImplementedPrefixes:  allowNotImplemented,
+	}
+
+	return api, nil
+}
+
+func (a *mockAPI) Get(ctx context.Context, o types.IdentifiedObject, opts ...types.GetOption) error {
+	identifier, basePath, err := a.prepareRequest(types.OperationGet, o)
+	if err != nil {
+		return err
+	}
+
+	if data, found := a.resourceStorage[basePath][identifier]; found {
+		err = json.Unmarshal(data, o)
+		return err
+	}
+
+	return api.ErrNotFound
+}
+
+func (a *mockAPI) List(ctx context.Context, o types.FilterObject, opts ...types.ListOption) error {
+	_, _, err := a.prepareRequest(types.OperationList, o)
+	if err != nil {
+		return err
+	}
+
+	return errors.New("List not implemented in mock API client")
```

Comment:
please always create a variable for errors to allow outside code to check if the returned error is a specific error - never `return errors.New(...)`